core/src/main/scala/tofu/Context.scala

```diff
@@ -80,66 +162,149 @@ object WithContext {
   def apply[F[_], C](implicit ctx: WithContext[F, C]): WithContext[F, C] = ctx
 }
 
-/** lightweight version of ApplicativeLocal
+/** Allows to run some computation with notion of altered context
   * consider using `WithLocal` for better type inference
   */
 trait Local[F[_]] extends Context[F] {
+
+  /** Alters context for computation
+    *
+    * @example
+    * Example of usage is to hide sensitive information {{{
+    *
+    * import tofu.syntax.context._
+    *
+    * case class UserContext(
+    *              id: String,
+    *              phoneNumber: String //very sensitive
+    *            )
+    * def hidePartOfPhoneNumber: String => String = ???
+    *
+    * def contextualLogInfo[F[_]](message: String)
+    *     (implicit hasUserContext: UserContext In F): F[Unit] =
+    *     ??? //logs both message AND UserContext
+    *
+    * def program[F[_]: Monad: ] = for {
+    *    user <- obtainUserSomehow[F]
+    *    implicit0(hasUserContext: User In F) <- Context.const[F, User](user.toContext)
+    *    logUser =  contextualLogInfo[F](s"Successfully obtained user")
+    *    _ <- logUser.local(hidePartOfPhoneNumber) //logs only part of phone number
+    * } yield ()
+    * }}}
+    * @param fa      computation that is going to run with different context
+    * @param project pure function
+    * @return result of ran computation `fa`
+    */
   def local[A](fa: F[A])(project: Ctx => Ctx): F[A]
 
+  /** Allows to focus [[Local]] on inner parts of its context with lens.
+    *
+    * @param contains lens that can extract from `Ctx` or set some value of type `A`
+    */
   def subcontext[A](contains: Ctx Contains A): WithLocal[F, A] = new LocalContainsInstance[F, Ctx, A](this, contains)
 }
 
 object Local {
   def apply[F[_]](implicit ctx: Local[F]): HasLocal[F, ctx.Ctx] = ctx
+
   type Aux[F[_], C] = HasLocal[F, C]
 }
 
-/** lightweight version of ApplicativeLocal
-  * allows to modify value of `C` locally for some subprocess
-  */
+/** Synonym for [[Local]] with C as Ctx for better type inference */
 trait WithLocal[F[_], C] extends Local[F] with WithContext[F, C]
 
 object WithLocal {
   def apply[F[_], C](implicit ctx: WithLocal[F, C]): WithLocal[F, C] = ctx
 }
 
+/** Allows to evaluate contextual computation with some context
+  *
+  * The main use case for it is to obtain some context `Lower[Ctx]`,
+  * and then to run another contextual computation F[A] that can use `F[Ctx]` inside.
+  *
+  * @tparam F context-aware effect e.g.`ReaderT[Lower, Ctx, *]`
+  */
 trait Provide[F[_]] extends ContextBase {
   type Ctx
+
+  /** Result effect type without any notion of context */
   type Lower[A]
 
+  /** Runs `fa` providing `ctx` to it.
```

Comment:
You can reuse example from Local, but instead of creating constant context, run `logUser` via `runContext` with user context